d2common/d2fileformats/d2ds1/layer.go

```diff
@@ -0,0 +1,129 @@
+package d2ds1
+
+// layerStreamType represents a layer stream type
+type layerStreamType int
+
+// Layer stream types
+const (
+	layerStreamWall1 layerStreamType = iota
+	layerStreamWall2
+	layerStreamWall3
+	layerStreamWall4
+	layerStreamOrientation1
+	layerStreamOrientation2
+	layerStreamOrientation3
+	layerStreamOrientation4
+	layerStreamFloor1
+	layerStreamFloor2
+	layerStreamShadow1
+	layerStreamSubstitute1
+)
+
+type tileRow []Tile     // index is x coordinate
+type tileGrid []tileRow // index is y coordinate
+
+type layer struct {
+	tiles tileGrid
+}
```

Comment:
lol, why did you make it as a two types? isn one not enough?

Reply:
eh, for clarity; "a grid is a slice of rows"